pkg/runctl/controller.go

```diff
@@ -206,23 +206,12 @@ func (c *Controller) processNextWorkItem() bool {
 }
 
 func (c *Controller) changeState(pipelineRun k8s.PipelineRun, state api.State) error {
-	start := time.Now()
-	oldState, err := pipelineRun.UpdateState(state)
+	err := pipelineRun.UpdateState(state)
 	if err != nil {
 		klog.V(3).Infof("Failed to UpdateState of [%s] to %q: %q", pipelineRun.String(), state, err.Error())
 		return err
 	}
 
-	end := time.Now()
-	elapsed := end.Sub(start)
-	c.metrics.ObserveUpdateDurationByType("UpdateState", elapsed)
```

Comment:
Not sure if we need this metric. It could be used to measure the Commit duration.

Reply:
done